critiquebrainz/db/users.py

```diff
@@ -70,3 +73,543 @@ def get_many_by_mb_username(usernames):
+
+    return rows
+
+
+def get_reviews(user_id, date='1-1-1970'):
+    """Get reviews by a user from a specified time
+
+    Args:
+        user_id(uuid): ID of the user,
+        date(datetime): Date from which reviews submitted(default: UTC)
+
+    Returns:
+        List of reviews by the user from the time specified
+        {
+            "id": (uuid)
+            "entity_id": (uuid),
+            "entity_type": event or place or release_group,
+            "user_id": (uuid),
+            "edits": (int),
+            "is_draft": (bool),
+            "license_id": (str),
+            "language": (str),
+            "source": (str),
+            "source_url" (str)
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+           SELECT *
+           FROM review
+      LEFT JOIN
+        (SELECT review_id
+              , min(timestamp)
+             AS creation_time
+           FROM revision
+       GROUP BY review_id)
+             AS review_create
+             ON review.id = review_id
+          WHERE user_id = :user_id
+            AND creation_time > :date
+        """), {
+            "user_id": user_id,
+            "date": date
+        })
+
+        rows = result.fetchall()
+        rows = [dict(row) for row in rows]
+
+    return rows
+
+
+def update(user, display_name=None, email=None, show_gravatar=None):
+    """Update info of a user
+
+    Args:
+        user: User object whose info is to be updated,
+        display_name(str): Display name of user,
+        email(str): Email of the user,
+        show_gravatar(bool): whether to show gravatar
+    """
+    if display_name is None:
+        display_name = user.display_name
+    if show_gravatar is None:
+        show_gravatar = user.show_gravatar
+    if email is None:
+        email = user.email
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            UPDATE "user"
+               SET display_name = :display_name,
+                  show_gravatar = :show_gravatar,
+                          email = :email
+             WHERE id = :user_id
+        """), {
+            "user_id": user.id,
+            "display_name": display_name,
+            "show_gravatar": show_gravatar,
+            "email": email
+        })
+
+
+def delete(user_id):
+    """Delete a user
+    Deletes user, votes, reviews, revisions and spam reports
+    by it
+
+    Args:
+        user_id(uuid): ID of the user to be deleted
+    """
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            DELETE
+              FROM "user"
+             WHERE id = :user_id
+        """), {
+            "user_id": user_id
+        })
+        connection.execute(sqlalchemy.text("""
+            DELETE
+              FROM "vote"
+             WHERE user_id = :user_id
+        """), {
+            "user_id": user_id
+        })
```

Comment:
Associated information shouldn't be deleted like this. See `ON DELETE` rules in https://www.postgresql.org/docs/9.5/static/ddl-constraints.html.